src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbol.cs

```diff
@@ -1285,7 +1285,8 @@ private ImmutableArray<ParameterSymbol> ComputeParameters(Binder binder, BasePro
                 }
                 else if ((object)_setMethod != null && param.Name == ParameterSymbol.ValueParameterName)
                 {
-                    diagnostics.Add(ErrorCode.ERR_DuplicateGeneratedName, _location, param.Name);
+                    // note: there can only ever be one location for param, as it cannot be used in partial for a property
+                    diagnostics.Add(ErrorCode.ERR_DuplicateGeneratedName, param.Locations[0], param.Name);
```

Comment:
Usually, symbols from source have at least one location. However, to have peace of mind you can write defensive code, i.e. something like ```param.Locations.FirstOrDefault() ?? _location```.